includes/blocks/class-sensei-block-contact-teacher.php

```diff
@@ -46,20 +46,72 @@ public function register_block() {
 	 */
 	public function render_contact_teacher_block( $attributes, $content ): string {
 		global $post;
-		$href = add_query_arg( array( 'contact' => $post->post_type ) );
 
-		return wp_kses_post( $this->add_href_attribute( $content, $href ) );
+		if ( ! empty( Sensei()->settings->settings['messages_disable'] ) || ! is_user_logged_in() ) {
+			return '';
+		}
+
+		$contact_form_link = add_query_arg( array( 'contact' => $post->post_type ) );
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		$message_sent = ( isset( $_GET['send'] ) && 'complete' === $_GET['send'] );
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		$contact_form_open = isset( $_GET['contact'] ) || $message_sent;
+
+		$contact_form = $this->teacher_contact_form( $post, $message_sent );
+
+		return '<div id="private_message" class="sensei-collapsible">
+				' . ( $this->add_button_attributes( $content, $contact_form_link ) ) . '
+				<div class="sensei-collapsible__content ' . ( $contact_form_open ? '' : 'collapsed' ) . '">' . $contact_form . '</div>
+			</div>';
+	}
+
+	/**
+	 * Render contact teacher form.
+	 *
+	 * @param WP_Post $post         The current post.
+	 * @param bool    $message_sent Display message sent feedback.
+	 *
+	 * @return string
+	 */
+	private function teacher_contact_form( $post, $message_sent ) {
+
+		$confirmation = '';
+		if ( $message_sent ) {
+
+			$confirmation_message = __( 'Your private message has been sent.', 'sensei-lms' );
+			$confirmation         = '<div class="sensei-message tick">' . esc_html( $confirmation_message ) . '</div>';
+		}
+		$nonce = wp_nonce_field( 'message_teacher', 'sensei_message_teacher_nonce', true, false );
+
+		return '
+			' . $confirmation . '
+			<form name="contact-teacher" action="" method="post" class="sensei-contact-teacher-form">
+				<label>' . esc_html__( 'Send Private Message', 'sensei-lms' ) . '</label>
+				<textarea name="contact_message" placeholder="' . esc_attr__( 'Enter your private message.', 'sensei-lms' ) . '"></textarea>
```

Comment:
WDYT to add a `required` to this field? If we send a message without a text it's submitting the page without feedback.

Reply:
Makes sense! 75d6654